### PR TITLE
Add types to EntityListenerResolver

### PR DESCRIPTION
**Pull request by derrabus**
Part of #9772

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/EntityListenerResolver.php [hunk 1]
```diff
@@ -25,12 +23,12 @@ public function clear($className = null);
...
-    public function resolve($className);
+    public function resolve(string $className): object;
 
     /**
      * Register a entity listener instance.
      *
      * @param object $object An entity listener
      */
-    public function register($object);
+    public function register(object $object): void;
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/EntityListenerResolver.php:29]: Do you want to keep the arguments `@param` docblock?

**Review by SenseException — DISMISSED** [review]

File: lib/Doctrine/ORM/Mapping/EntityListenerResolver.php (hunk 1 above)
Comment (derrabus) [reply lib/Doctrine/ORM/Mapping/EntityListenerResolver.php:29]: Not necessarily. I was unsure if it still provides value. What do you think? Remove?

Comment (SenseException) [reply lib/Doctrine/ORM/Mapping/EntityListenerResolver.php:29]: Yes, it can be removed.

Comment (derrabus) [reply lib/Doctrine/ORM/Mapping/EntityListenerResolver.php:29]: Done.

**Review by SenseException — DISMISSED** [review]

**Pull request merged**